packages/mobile/src/services/offline-downloader/offline-download-queue.ts

```diff
@@ -79,13 +138,52 @@ export const startDownloadWorker = async () => {
   queue.start()
 }
 
+export const cancelQueuedCollectionDownloads = async (
+  payloadsToCancel: CollectionDownloadWorkerPayload[]
+) => {
+  const payloadsToCancelById = groupBy(
+    payloadsToCancel,
+    (payload) => payload.collectionId
+  )
+  queue.stop()
+  const jobs = await queue.getJobs()
+  const jobsToCancel = jobs.filter(({ workerName, payload }) => {
+    try {
+      const parsedPayload: CollectionDownloadWorkerPayload = JSON.parse(payload)
+      return (
+        workerName === COLLECTION_DOWNLOAD_WORKER &&
+        (payloadsToCancelById[parsedPayload.collectionId] ?? []).some(
+          (payloadToCancel) => isEqual(payloadToCancel, parsedPayload)
+        )
+      )
+    } catch (e) {
+      console.warn(e)
+      return false
+    }
+  })
+  jobsToCancel.forEach(async (rawJob) => {
+    try {
+      const parsedPayload: CollectionDownloadWorkerPayload = JSON.parse(
+        rawJob.payload
+      )
+      rawJob.active ? queue.cancelJob(rawJob.id) : queue.removeJob(rawJob)
+
+      store.dispatch(removeCollectionDownload(parsedPayload))
+    } catch (e) {
+      console.warn(e)
+    }
+  })
+  queue.start()
+}
+
 export const cancelQueuedDownloads = async (
   payloadsToCancel: TrackDownloadWorkerPayload[]
 ) => {
   const payloadsToCancelById = groupBy(
     payloadsToCancel,
     (payload) => payload.trackId
   )
+  queue.stop()
```

Comment:
why move this up? just to make sure we don't accidentally push something in after we've canceled?

Reply:
it's so we don't let the workers pick up new jobs while we're deciding which ones to cancel (the inactive ones still in the queue)